### PR TITLE
ci/docs/build: repo infrastructure

### PR DESCRIPTION
**Pull request by DaniBodor**
I would recommend reviewing this one commit at a time (and skipping the first commit - the second one shows the changes compared to deeprank's dev readme).

Note that the action for creating pypi (test-) releases was created here, but the package is not yet on PyPi. How shall we proceed with this?

Because I added the citation file here with details for the different authors that I sought online, it would be good if all authors reviewed it. @gcroci2 , I will tag you for review of this PR first and if you approve, then I can ask the other authors to at least check their presonal data there before merging.
(or do you think it's better to merge and then create a separate issue asking them to confirm their details?)

fix: #72 and #73

### CODE REVIEW

**Review by gcroci2 — APPROVED** [review]
I added (very) minor suggestions, and commented on the citation file. We should change the base of this PR to `dev` before merging, right?

For TestPyPi, I've created myself an entry for the package, which will be published as soon as we run for the first time the action. 

Regarding the PyPi package, @nacloos has created already an entry for the project (https://pypi.org/project/neurogym/). Could you add us as owners on PyPi? So we can do edits and add tokens if needed

File: CITATION.cff [hunk 1]
```diff
@@ -0,0 +1,68 @@
+# This CITATION.cff file was generated with cffinit.
+# Visit https://bit.ly/cffinit to generate yours today!
+
+cff-version: 1.2.0
+title: neurogym
+message: >-
+  If you use this software, please cite it using the
+  metadata from this file.
+type: software
+authors:
```
[the remark this answers is not in the document]
Comment (DaniBodor) [reply CITATION.cff:10]: Thanks for pointing this out, I hadn't realized. I now update the CFF to follow the zenodo file. I will stick to that order of authors of authors for now as well, and we can update that later if need be.
I did notice that @nacloos is not part of the zenodo file. I'm pretty sure they should be added as an author of the package. Should I add them to the zenodo as well?

File: README.md [hunk 2]
```diff
@@ -86,15 +107,5 @@ class YourTask(ngym.PeriodEnv):
...
-    [Manuel Molano](https://github.com/manuelmolano) (manuelmolanomazon@gmail.com).
-    [Guangyu Robert Yang](https://github.com/gyyang) (gyyang.neuro@gmail.com).
-
-* Contributors (listed in chronological order)
 
-    [Marta Fradera](https://github.com/martafradera),
-    [Jordi Pastor](https://github.com/pastorjordi),
-    [Jeremy Forest](https://github.com/jeremyforest),
-    [Ru-Yuan Zhang](https://github.com/ruyuanzhang)
+Please see our [citation file](CITATION.cff) for a list of authors and contributors
```
[the remark this answers is not in the document]
Comment (DaniBodor) [reply README.md:111]: all of them are in the zenodo/CFF

File: CITATION.cff (hunk 1 above)
[the remark this answers is not in the document]
Comment (gcroci2) [reply CITATION.cff:10]: He's there (see [the zenodo file in the dev branch](https://github.com/neurogym/neurogym/blob/dev/.zenodo.json)); please consider this file's version as the "correct" one in terms of people listed; who is not in this zenodo file's version, should not be there (but should be listed on the readme)

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (gcroci2) [reply README.md:111]: they shouldn't be all there, see my comment above

[the remark this answers is not in the document]
Comment (DaniBodor) [reply README.md]: yeah, my bad. hadn't realized that the zenodo had changed. As for the readme: Nathan is listed as a contact on main branch but not on dev branch. Which is correct?

[the remark this answers is not in the document]
Comment (DaniBodor) [reply README.md]: > We could refer to the zenodo/cff files for the authors and then leave here the contributors' names.

I did this in the end, so the point is moot

**Pull request merged**